pubsub/query/parser_test.go

```diff
@@ -0,0 +1,54 @@
+package query_test
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/tendermint/tmlibs/pubsub/query"
+)
+
+// TODO: fuzzy testing?
+func TestParser(t *testing.T) {
+	cases := []struct {
+		query string
+		valid bool
+	}{
+		{"tm.events.type=NewBlock", true},
+		{"tm.events.type = NewBlock", true},
+		{"tm.events.type>NewBlock", false},
+		{"", false},
+		{"=", false},
+		{"=NewBlock", false},
+		{"tm.events.type=", false},
+
+		{"tm.events.typeNewBlock", false},
+		{"NewBlock", false},
+		{"", false},
+
+		{"tm.events.type=NewBlock AND", false},
+		{"AND tm.events.type=NewBlock ", false},
+		{"tm.events.type=NewBlock AND abci.account.name=Igor", true},
+
+		{"abci.account.name CONTAINS Igor", true},
+
+		{"tx.date > DATE 2013-05-03", true},
+		{"tx.date < DATE 2013-05-03", true},
+		{"tx.date <= DATE 2013-05-03", true},
+		{"tx.date >= DATE 2013-05-03", true},
+
+		{"tx.date > TIME 2013-05-03T14:45:00+07:00", true},
+		{"tx.date < TIME 2013-05-03T14:45:00-02:00", true},
+		{"tx.date <= TIME 2013-05-03T14:45:00Z", true},
+		{"tx.date >= TIME 2013-05-03T14:45:00Z", true},
+	}
+
+	for _, c := range cases {
+		_, err := query.New(c.query)
+		if c.valid {
+			assert.NoError(t, err, fmt.Sprintf("Query was '%s'", c.query))
+		} else {
+			assert.Error(t, err, fmt.Sprintf("Query was '%s'", c.query))
```

Comment:
assert.NotNil(t, err, "Query '%s'", c.query)

These are the forms usually used in other test cases.  Sprintf is implied with the assert, if there are extra arguments